From c3609263094816eee96bef416f7ed72b49c8f2b8 Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Wed, 15 Jan 2014 16:51:18 +0100
Subject: [PATCH 05/10] tests: Make btrfs recognize devices from tests suite
 properly

This is ugly hack to fix a problem with test suite and btrfs
where ?sometimes? btrfs prints out device name in the path
of the test suite rather than path in the real '/dev/'
directory. This should cover that without any impact on
real usage.

Signed-off-by: Lukas Czerner <lczerner@redhat.com>
---
 ssmlib/backends/btrfs.py | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/ssmlib/backends/btrfs.py b/ssmlib/backends/btrfs.py
index b238d52..836f108 100644
--- a/ssmlib/backends/btrfs.py
+++ b/ssmlib/backends/btrfs.py
@@ -119,6 +119,13 @@ class Btrfs(template.Backend):
                 fs_used = float(misc.get_real_size(array[6]))
 
             elif array[0] == 'devid':
+                # This is ugly hack to fix a problem with test suite and btrfs
+                # where ?sometimes? btrfs prints out device name in the path
+                # of the test suite rather than path in the real '/dev/'
+                # directory. This should cover that without any impact on
+                # real usage
+                if not os.path.islink(array[7]):
+                    array[7] = re.sub(r'.*/dev/', '/dev/', array[7])
                 dev['dev_name'] = misc.get_real_device(array[7])
 
                 if not pool_name:
-- 
1.8.3.1

